src/ScottLogic.Internal.Training/ScottLogic.Internal.Training.Matcher/OrderMatcher.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace ScottLogic.Internal.Training.Matcher
+{
+    public class OrderMatcher
```

Comment:
The methods in this file are quite complicated, and difficult to test as there are a lot of paths through the code. I would extract parts out where you can to make things simpler. Take a shot at it first, then call me over and we can go through it together.